packages/broccoli-eyeglass/src/broccoli_sass_compiler.ts

```diff
@@ -627,26 +627,33 @@ export default class BroccoliSassCompiler extends BroccoliPlugin {
    * @return hash object of the file data
    **/
   hashForFile(absolutePath: string): string {
-    let cacheKey = `hashForFile(${absolutePath})`;
-    let cachedHash = this.buildCache.get(cacheKey);
-    if (cachedHash) {
-      return cachedHash as string;
+    return this.fileKey(absolutePath);
+  }
+
+  /* compute a key for a file that will change if the file has changed. */
+  fileKey(file: string, isRealPath = false): string {
+    let cachedKeyKey = `fileKey(${file})`;
+    let cachedKey = this.buildCache.get(cachedKeyKey) as string;
+    if (cachedKey) { return cachedKey; }
+    let stat = statSync(file);
+    let key;
+    if (!isRealPath && stat.isSymbolicLink) {
+      key = this.fileKey(realpathSync(file), true);
     } else {
-      let data = fs.readFileSync(absolutePath, "UTF8");
-      let hash = crypto.createHash("md5").update(data).digest("hex");
-      this.buildCache.set(cacheKey, hash);
-      return hash;
+      key = `${stat.mtimeMs}:${stat.size}:${stat.mode}`;
```

Comment:
I think using `mtimeMs` is a breaking change, as it was introduced in node 8.10 https://github.com/nodejs/node/commit/157fc28710178e328bd6d3a66c33e2649ba81155#diff-acabf706a8aa070a8796e3573f7e4678R556 but this and ember-cli-eyglass [still support node >= 6](https://github.com/linkedin/eyeglass/blob/b130d929d28a3ffcb5a688a855f2a5cab6393e64/packages/broccoli-eyeglass/package.json#L63).

I think it's good to use, and may justify the major bump. Especially as Node 6 is technically supposed to EOL shortly.

Other upsides of that would be that you can start using native `async/await` in your code.

So options as i see:

* keep support for node 6, and fallback to `mtime` if `mtimeMs` is missing
* drop support for node 6, and do a major bump. (node 6 is EOL shortly anyways)

Reply:
@stefanpenner Thanks for pointing this out. I'll write a shim that uses it if available.

Reply:
@chriseppstein perfect. I suspect other parts of the broccoli/build ecosystem would be better served by also switching to `mtimeMs`. So thanks for surfacing this, I totally wasn't aware it was a feature.

Once node 8 is dead, I'll explore utilize it across the board.